trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/library/AtomicLibraryModel.java

```diff
@@ -480,7 +493,29 @@ private ExpressionResult applyMemoryOrders(final ILocation loc, final Expression
 
 	@Override
 	public Collection<TypeModel> getTypeModels() {
-		// TODO: Handle types like atomic_int etc. here
-		return List.of();
+		return List.of(new TypeModel("atomic_bool", CPrimitive.constructAtomicType(CPrimitives.BOOL)),
+				new TypeModel("atomic_char", CPrimitive.constructAtomicType(CPrimitives.CHAR)),
+				new TypeModel("atomic_schar", CPrimitive.constructAtomicType(CPrimitives.SCHAR)),
+				new TypeModel("atomic_uchar", CPrimitive.constructAtomicType(CPrimitives.UCHAR)),
+				new TypeModel("atomic_short", CPrimitive.constructAtomicType(CPrimitives.SHORT)),
+				new TypeModel("atomic_ushort", CPrimitive.constructAtomicType(CPrimitives.USHORT)),
+				new TypeModel("atomic_int", CPrimitive.constructAtomicType(CPrimitives.INT)),
+				new TypeModel("atomic_uint", CPrimitive.constructAtomicType(CPrimitives.UINT)),
+				new TypeModel("atomic_long", CPrimitive.constructAtomicType(CPrimitives.LONG)),
+				new TypeModel("atomic_ulong", CPrimitive.constructAtomicType(CPrimitives.ULONG)),
+				new TypeModel("atomic_llong", CPrimitive.constructAtomicType(CPrimitives.LONGLONG)),
+				new TypeModel("atomic_ullong", CPrimitive.constructAtomicType(CPrimitives.ULONGLONG)),
+				new TypeModel("memory_order",
+						new CEnum("memory_order",
+								Arrays.stream(MemoryOrder.values()).sorted(Comparator.comparing(MemoryOrder::getValue))
```

Comment:
Regarding the discussion of how to model this enum: I am a bit confused now -- what role does the order of the names in the `CEnum` instance play? As you do provide constant models, I guess the fields are not implicitly numbered from `1` to `n`. Is the order at all relevant?

Reply:
Oh, you are right we actually don't care about the fields or their order at all in `CEnum`, I somehow confused this with `CStructOrUnion` (where we lookup the fields, but the order still does not matter).
Instead of an `enum`, we could define `memory_order` also using just `int`, but I am not sure if that is better.